operators/src/pro/cache/shared_cache.rs

```diff
@@ -0,0 +1,1190 @@
+            }),
+        })
+    }
+}
+
+impl TestDefault for SharedCache {
+    fn test_default() -> Self {
+        Self {
+            backend: RwLock::new(CacheBackend {
+                vector_caches: Default::default(),
+                raster_caches: Default::default(),
+                lru: LruCache::unbounded(), // we need no cap because we evict manually
+                cache_size: CacheSize::new(usize::MAX),
+                landing_zone_size: CacheSize::new(usize::MAX),
+            }),
+        }
+    }
+}
+
+/// Holds all the cached results for an operator graph (workflow)
+#[derive(Debug, Default)]
+pub struct OperatorCacheEntry<C, L> {
+    // for a given operator and query we need to look through all entries to find one that matches
+    // TODO: use a multi-dimensional index to speed up the lookup
+    entries: HashMap<CacheEntryId, C>,
+
+    // running queries insert their tiles as they are produced. The entry will be created once the query is done.
+    // The query is identified by a Uuid instead of the query rectangle to avoid confusions with other queries
+    landing_zone: HashMap<QueryId, L>,
+}
+
+impl<C, L> OperatorCacheEntry<C, L> {
+    pub fn new() -> Self {
+        Self {
+            entries: Default::default(),
+            landing_zone: Default::default(),
+        }
+    }
+
+    fn insert_landing_zone_entry(
+        &mut self,
+        query_id: QueryId,
+        landing_zone_entry: L,
+    ) -> Result<(), CacheError> {
+        let old_entry = self.landing_zone.insert(query_id, landing_zone_entry);
+
+        if old_entry.is_some() {
+            Err(CacheError::QueryIdAlreadyInLandingZone)
+        } else {
+            Ok(())
+        }
+    }
+
+    fn remove_landing_zone_entry(&mut self, query_id: &QueryId) -> Option<L> {
+        self.landing_zone.remove(query_id)
+    }
+
+    fn landing_zone_entry_mut(&mut self, query_id: &QueryId) -> Option<&mut L> {
+        self.landing_zone.get_mut(query_id)
+    }
+
+    fn insert_cache_entry(
+        &mut self,
+        cache_entry_id: CacheEntryId,
+        cache_entry: C,
+    ) -> Result<(), CacheError> {
+        let old_entry = self.entries.insert(cache_entry_id, cache_entry);
+
+        if old_entry.is_some() {
+            Err(CacheError::CacheEntryIdAlreadyInCache)
+        } else {
+            Ok(())
+        }
+    }
+
+    fn remove_cache_entry(&mut self, cache_entry_id: &CacheEntryId) -> Option<C> {
+        self.entries.remove(cache_entry_id)
+    }
+
+    fn is_empty(&self) -> bool {
+        self.entries.is_empty() && self.landing_zone.is_empty()
+    }
+
+    fn iter(&self) -> impl Iterator<Item = (&CacheEntryId, &C)> {
+        self.entries.iter()
+    }
+}
+
+identifier!(QueryId);
+
+impl ByteSize for QueryId {}
+
+identifier!(CacheEntryId);
+
+impl ByteSize for CacheEntryId {}
+
+/// Holds all the tiles for a given query and is able to answer queries that are fully contained
```

Comment:
Holds all the tiles OR CHUNKS, rihgt?

Reply:
f768513ba509945b3b2369cf8409f7751fef435b

Reply:
[f768513](https://github.com/geo-engine/geoengine/pull/833/commits/f768513ba509945b3b2369cf8409f7751fef435b)